### DISCUSSION TITLE
[Suggestion] Add Services Ports to LXC Descriptions

### DISCUSSION THREAD

**Discussion by snowy-jaguar** [tteck/Proxmox#3561] [Ideas]
If a LXC runs a service, like Portainer or one of the *arrs, then add a note in the LXC description declaring the port that service will be running on. Somehting like:
```
# This function sets the description of the container.
description() {
  IP=$(pct exec "$CTID" ip a s dev eth0 | awk '/inet / {print $2}' | cut -d/ -f1)
  pct set "$CTID" -description "<div align='center'><a href='https://Helper-Scripts.com' target='_blank' rel='noopener noreferrer'><img src='https://raw.githubusercontent.com/tteck/Proxmox/main/misc/images/logo-81x112.png'/></a>

  # ${APP} LXC
${APP} is available on port ${port}
  <a href='https://ko-fi.com/proxmoxhelperscripts'><img src='https://img.shields.io/badge/&#x2615;-Buy me a coffee-blue' /></a>
  </div>"
  if [[ -f /etc/systemd/system/ping-instances.service ]]; then
    systemctl start ping-instances.service
  fi
}
```
And if the LXC has multiple services then it could be
```
# This function sets the description of the container.
description() {
  IP=$(pct exec "$CTID" ip a s dev eth0 | awk '/inet / {print $2}' | cut -d/ -f1)
  pct set "$CTID" -description "<div align='center'><a href='https://Helper-Scripts.com' target='_blank' rel='noopener noreferrer'><img src='https://raw.githubusercontent.com/tteck/Proxmox/main/misc/images/logo-81x112.png'/></a>

  # ${APP} LXC
${APP} is available on port ${port}
  <a href='https://ko-fi.com/proxmoxhelperscripts'><img src='https://img.shields.io/badge/&#x2615;-Buy me a coffee-blue' /></a>
  </div>"
  if [[ -f /etc/systemd/system/ping-instances.service ]]; then
    systemctl start ping-instances.service
  fi
}
```
Theres are just mockups of how I think it could be, I have done no testing to find out if this is useable code.

**Comment by tteck** [comment]
Where is the `port=` variable? Every script would need to be modified to include that variable, or you'd encounter an error. It's not worth the effort for me, especially since you can simply include any additional information in the notes.

**Reply by snowy-jaguar** [reply]
That's what I was trying to say in my last sentence, I've done no actual testing or deep look in to how the script would need to be adapted for the `ports=` variable. Those were simple mockups of how I should it could be displayed in the LXC description. I'll take a poke around  in a few weeks and see if I can get a `ports` variable implemented.
